商品一覧表示機能

SymTr opened the pull request:
# What 
ブランチを作成する。
商品一覧表示機能を実装する。
「売却済みの商品は、画像上に『sold out』の文字が表示されること」という機能に関しては、codingは済みです。商品購入機能実装後にデモで確認いたします。

# Why
商品一覧表示機能実装のため


# GYAZO
* 商品のデータがない場合は、ダミー商品が表示されている動画
https://gyazo.com/12561642956f31ff2370eb5993cf8da2

*  商品のデータがある場合は、商品が一覧で表示されている動画
https://gyazo.com/8e3f1f74691e168ae5efdf9b75b1ccfe
 
注）カラムの並び順はわかりやすいように作成日を移動しました。
  撮影後は元の位置にもどしました。

SymTr committed:
ダミー商品表示設定

SymTr committed:
出品が新しい順に並べる設定

SymTr committed:
商品の4項目表示設定

SymTr committed:
[未完成] Sold Out表示設定　Purchaseモデル作成前に一旦commit

SymTr committed:
Purchase model creation

SymTr committed:
purchase table creation and  SOLD OUT on #index setting done!!

mizuki-ueda commented on app/views/items/index.html.erb:
```diff
@@ -129,24 +129,24 @@
+     <% @items.each do |item|%>
       <li class='list'>
-        <%= link_to "#" do %>
+        <%= link_to item_path(item) do %>
```
```
<%= link_to item_path(item) do %>
```
詳細ページへのパスは、次の商品詳細機能ブランチでの実装内容になるため、本ブランチでは元の'#'のままにしましょう。

mizuki-ueda commented on app/views/items/index.html.erb:
```diff
@@ -129,24 +129,24 @@
     <ul class='item-lists'>
 
       <%# 商品のインスタンス変数になにか入っている場合、中身のすべてを展開できるようにしましょう %>
```
```
<%# 商品のインスタンス変数になにか入っている場合、中身のすべてを展開できるようにしましょう %>
```
「〜しましょう」で終わる実装案内コメントは実装完了後に削除しましょう。
企業提出時に実装案内コメントが残っていないようにするためです。
(他にも実装案内コメントがあれば削除しておきましょう)

mizuki-ueda commented on app/views/items/index.html.erb:
```diff
@@ -129,24 +129,24 @@
-
-          <%# 商品が売れていればsold outを表示しましょう %>
+          <%= image_tag item.image, class: "item-img" if item.image.attached? %>
+          <% if item.purchase.present? %>
```
```
<% if item.purchase.present? %>
```
購入機能が未実装の状態では売り切れ表示の実装は動作確認ができないため、売り切れ表示の実装は元に戻すかコメントアウトしておきましょう。
（現状は全ての商品にsold outが表示される状態で問題ありません）

mizuki-ueda reviewed (commented):
レビュー依頼ありがとうございます！

コメントいたしましたので、ご確認お願いいたします。
修正後は、再度レビュー依頼フォームのご提出をお願いいたします。

SymTr committed:
[fix] 詳細ページパスを削除、sold out をコメントアウト

1) 詳細ページへのパスは、次の商品詳細機能ブランチでの実装内容になるため、本ブランチでは元の'#'のままにした

2) 購入機能が未実装の状態では売り切れ表示の実装は動作確認ができないため、売り切れ表示の実装は元に戻すかコメントアウト

SymTr commented:
ご連絡ありがとうございます。
ご指摘の箇所を修正いたしました。

再度ご確認をお願いいたします。

mizuki-ueda reviewed (commented):
レビュー依頼ありがとうございます！

LGTMとさせていただきます！
実装、お疲れ様でした！

マージできましたら、ローカルのmainブランチへ、忘れずにプルしましょう。
そのあと、新しいブランチを作成し、次の実装に取り組んでください。

引き続き、実装頑張っていきましょう。
応援しております！
